WordPress/wordpress.org
# Plugin Directory: Use the shared favorite button for favoriting plugins

This updates the single plugin page to use the Favorite block, and so that the buttons lined up correctly I've also switched the plugin action buttons (download, preview) to use button blocks as well. Download is now primary, with preview as an outline button, to prevent double primary buttons.

The focus state was also a little rough on the favorite block, so I've copied in the default focus styles from themes & patterns, which does update the style in the content as well (videos at the end to show before/after).

| | Before | After |
|---|---|---|
| Favorited | ![](https://github.com/user-attachments/assets/5aceab5d-a2ef-4e25-aff6-21212ad8db6f) | ![](https://github.com/user-attachments/assets/35c71e81-803f-4ce9-8283-55be8751ed14) |
| Unfavorited | ![](https://github.com/user-attachments/assets/a398b037-8ecc-4437-92ce-69e87b5f7768) | ![](https://github.com/user-attachments/assets/49c653e5-50e5-4ce0-b97f-c656e83dddff) |

Focus states, before

https://github.com/user-attachments/assets/0e70b899-eae4-4bca-bcc5-1af0f3355621

Focus states, after

https://github.com/user-attachments/assets/b6665ad6-87c2-478c-8d26-a23a80be3d9e


**To test**

- Use a sandbox, be logged in
- Visit any plugin page
- The favorite button should appear next to the download button
- Click the button, it will trigger an API request but not reload the page
- The button should update to reflect the new favorite status
- Tab through the site, the focus state should be visible and match the text color
    - No visible focus on the screenshots or FAQs, but that's true on production too
- Download and Preview buttons should still work

If you're not logged in, the button is not visible.
